### PR TITLE
add backward-compatibility-check workflow

### LINKED ISSUE #105: Introduce  Roave/BackwardCompatibilityCheck as CI job

### LINKED ISSUE #111: Add Roave/BackwardCompatibilityCheck as CI Job
https://github.com/Roave/BackwardCompatibilityCheck

### PR DESCRIPTION
**Pull request by DavidBadura**
fix #105

### CODE REVIEW

**Comment by DanielBadura** [comment]
I think the should push out the tools like this into a separate `composer.json` since alot of deps are getting pulled in which the user does not need. The SA tools (psalm, phpstan, phpcs) could stay in the normal json i think. WDYT @DavidBadura ?

**Comment by DavidBadura** [comment]
Yes, let's do it that way, we can simply adjust it again if it doesn't fit :)

File: .github/workflows/backward-compatibility-check.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        with:
+          dependency-versions: ${{ matrix.dependencies }}
+
+      - uses: "ramsey/composer-install@v2"
+        with:
+          dependency-versions: ${{ matrix.dependencies }}
+          working-directory: 'tools'
+
+      - name: "Check for BC breaks"
+        run: "tools/vendor/bin/roave-backward-compatibility-check --format=github-actions --install-development-dependencies"
```
Comment (DanielBadura) [inline .github/workflows/backward-compatibility-check.yml:50]: not quite sure if we need the dev deps here..

**Comment by DavidBadura** [comment]
LGTM

**Pull request merged**